src/interface_rpc_url_provider.py

```diff
@@ -17,17 +18,23 @@ def __init__(self, charm, relation_name):
             charm.on[relation_name].relation_joined, self._on_relation_joined
         )
 
-    def _on_relation_joined(self, event: RelationChangedEvent) -> None:
+    def _on_relation_joined(self, event: RelationJoinedEvent) -> None:
         """This event is used to send the ws or http rpc url to another client."""
 
-        service_args_obj = ServiceArgs(self._charm.config.get('service-args'))
-        ingress_address = event.relation.data.get(self.model.unit)['ingress-address']
-        if service_args_obj.ws_port:
-            url = f'ws://{ingress_address}:{service_args_obj.ws_port}'
-        elif service_args_obj.rpc_port:
-            url = f'http://{ingress_address}:{service_args_obj.rpc_port}'
-        else:
+        service_args_obj = ServiceArgs(self._charm.config.get('service-args'), "")
+
+        ws_port = service_args_obj.ws_port
+        rpc_port = service_args_obj.rpc_port
+        if not ws_port and not rpc_port:
             event.defer()
             return
 
-        event.relation.data[self.model.unit]['url'] = url
+        # In newer version of Polkadot the ws options are removed, and ws and http uses the same port specified by --rpc-port instead.
+        if "--ws-port" not in utils.get_client_binary_help_output():
+            ws_port = rpc_port
```

Comment:
Would this be reason enough for printing something to the Juju debuglog as well perhaps?

Reply:
Would not hurt. I don't have the logger there though. You know if we have done logging from the interfaces in some other charm? Not sure if I should try accessing the logger in `charm.py` or instantiate a new logger...

Reply:
Maybe just do `logger = logging.getLogger(__name__)` , I guess it returns the same logger...

Reply:
It worked.